recipe/meta.yaml

```
@@ -1,57 +1,127 @@
 {% set name = "pennylane" %}
-{% set version = "0.23.0" %}
+{% set version = "0.29.0" %}
+
 
 package:
-  name: {{ name|lower }}
+  name: pennylane-suite
   version: {{ version }}
 
 source:
-  url: https://github.com/PennyLaneAI/pennylane/archive/v{{ version }}.tar.gz
-  sha256: b2d37227e098d4fc58556a07ba76eb5e2cefef52bff5605b465a01e5acb6941c
+  - url: https://github.com/PennyLaneAI/pennylane/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: e59e042174abd8f1596034d3f4dd113b24d06e4aadb5f2e5828fa3411fabf0df
+    folder: pennylane
+  - url: https://github.com/PennyLaneAI/pennylane-lightning/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: da9912f0286d1a54051cc19cf8bdbdcd732795636274c95f376db72a88e52d85
+    folder: pennylane-lightning
 
 build:
-  entry_points:
-    - pl-device-test=pennylane.devices.tests:cli
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
   number: 0
 
 requirements:
   host:
     - pip
-    - python >=3.7
-  run:
-    - appdirs
-    - autograd
-    - autoray
-    - cachetools
-    - networkx
-    - numpy
-    - pennylane-lightning >=0.23
-    - python >=3.7
-    - retworkx
-    - scipy
-    - semantic_version ==2.6
-    - toml
-
-test:
-  imports:
-    - pennylane
-  commands:
-    - pip check
-    - pl-device-test --help
-  requires:
-    - pip
-    - pytest
+    - python
+
+outputs:
+  - name: pennylane-core
+    build:
+      skip: true  # [win]
+      run_exports:
+        - {{ pin_subpackage('pennylane-core', max_pin='x.x.x') }}
+      entry_points:
+        - pl-device-test=pennylane.devices.tests:cli
+      script: cd "${SRC_DIR}/pennylane" && {{ PYTHON }} -m pip install . --no-deps -vv
+    requirements:
+      build:
+        - python
+      host:
+        - pip
+        - python
+      run:
+        - appdirs
+        - autograd
+        - autoray ==0.3.1
+        - cachetools
+        - cvxopt
+        - cvxpy
+        - networkx
+        - numpy <1.24
+        - opt_einsum
+        - python
+        - requests
+        - retworkx
+        - scipy
+        - semantic_version >=2.10
+        - toml
+
+  - name: pennylane-lightning-core
+    build:
+      skip: true  # [win]
+      run_exports:
+        - {{ pin_subpackage('pennylane-lightning-core', max_pin='x.x.x') }}
+      script: cd "${SRC_DIR}/pennylane-lightning" && {{ PYTHON }} -m pip install . --no-deps -vv
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        # Needed for Windows
+        - cmake
+        - ninja
+        - make
+      host:
+        - libblas
+        - liblapacke
+        - libgomp      # [linux]
+        - llvm-openmp  # [osx]
+        - pip
+        - pybind11
+        - python
+        - setuptools <=64.0.2 # version requirement from cvxpy
+      run:
+        - ninja
+        - numpy
+        - python
```

@conda-forge/pennylane I'm trying to debug this issue: https://github.com/conda-forge/pennylane-lightning-gpu-feedstock/issues/1#issuecomment-1494761395, and I guess this is the reason:
```suggestion
      run:
        - ninja
        - numpy
        - python
      run_constrained:
        - pennylane-lightning <=0.24.0
```
Without introducing a conflict between the old/new formats, the conda solver would take both formats into account, and depending on the solver internal (conda vs mamba) either format could be picked.

---

(fixed the upper bound lol)

---

I'm not familiar with this command. So this says that `pennylane-lightning-core` may depend on `pennylane-lightning`, but only versions `<0.0a0`, which are none of them. So it's not possible to depend on both.

---

Exactly 🙂 This is a common trick to ensure that two packages cannot coexist in the same environment (it's a solver conflict -> error). We used it too: https://github.com/conda-forge/cuquantum-feedstock/blob/0685dca47744279676ad40fb518e575d47c8aade/recipe/meta.yaml#L168-L172. There're other ways to do this, but this is the simplest.

---

I'll make a PR for this, unless you were just about to open one?

---

Would you mind doing it for me, Vincent? I don't have maintainer access here, so I can't just ask the bot to open a draft PR for me to edit 😅